src/Registration/cReg/NiftiImageData.cpp

```diff
@@ -102,7 +139,11 @@ NiftiImageData<dataType>& NiftiImageData<dataType>::operator=(const ImageData& t
             // Always float
             this->set_up_data(NIFTI_TYPE_FLOAT32);
             // Finally, copy the data
-            this->copy(to_copy.begin(), this->begin(), this->end());
+            auto &it_src = to_copy.begin();
+            auto &it_dst = this->begin();
+            for (; it_src != to_copy.end(); ++it_src, ++it_dst)
+                *it_dst = (*it_src).complex_float().real();
```

Comment:
this is truncating `complex` to `float`. 2 remarks:
- This would be amazingly confusing if `dataType` is complex. I know you're saying this can currently not happen, but do we guarantee it anywhere?
- If `dataType` is `float` or `double`, then this in my opinion shouldn't be in an assignment operator.

Reply:
No, `dataType` cannot be complex - see 12 data types dealt with by `change_datatype`.

As the destination can only store `float` data while the source has `complex_float_t` data, some kind of truncation (either `abs` or `real`) is unavoidable I am afraid.